src/app/shared/http-services/auth.http-service.ts

```diff
@@ -20,21 +20,23 @@ const longAuthServicesTimeout = 10000;
 export class AuthHttpService {
 
   private http: HttpClient; // an http client specific to this class, skipping all http interceptors
-  private cookieParams = new HttpParams({ fromObject: appConfig.authType === 'cookies' ? {
+  private cookieParams = appConfig.authType === 'cookies' ? {
     use_cookie: '1',
     cookie_secure: appConfig.secure ? '1' : '0',
     cookie_same_site: appConfig.sameSite ? '1' : '0',
-  } : {} });
+  } : undefined;
```

Comment:
`{}`?

I supposed you have tested that `...undefined` works as expected.

Reply:
Yes, it does and that works for any falsy values actually. But only for objects, with arrays it breaks.
Using the empty object is invalid for TypeScript because `HttpParams` `fromObject` option expects either an undefined value, either a populated object. :shrug: